### DISCUSSION TITLE
Saving a mesh as a binary 3d mask / numpy array

### DISCUSSION THREAD

**Discussion by helwilliams** [pyvista/pyvista#1888] [Q&A] [answered]
## Description

I am looking into using pyvista to create a 3D mesh from a set of surface coordinates. This is possible, however, I then want to convert the mesh to a 3D binary mask for further training. I haven't been able to see this function in the documentation so thought I would ask in the support section.

I would like the 3D numpy array of the closed surface and to be able to save it as a 'nii.gz' file for example.

Thanks in advance

**Comment by banesullivan** [comment, marked as answer]
Take a look at this example on my website: https://banesullivan.com/pyvista/examples/fault_mask.html

It uses a surface mesh to create a binary mask inside a volume

![sphx_glr_fault_mask_002](https://user-images.githubusercontent.com/22067021/143785849-07d8b8a8-ca4f-412e-8fcc-c7cf3ae4e18a.png)
